homeworks/david.tereshchenko_DavidTereshchenko/7-arrays-loops/arrays.js

```diff
@@ -0,0 +1,103 @@
+const arrayOne = [3, 0, -5, 1, 44, -12, 3, 0, 0, 1, 2, -3, -3, 2, 1, 4, -2, -3, -1];
+const arrayTwo = [-1, -8, -2];
+const arrayThree = [1, 7, 3];
+const arrayFour = [1, undefined, 3, 5, -3];
+const arrayFive = [1, NaN, 3, 5, -3];
+
+function maxNumberOfArr(array) {
+    let max = array[0];
+    for (let i = 1; i < array.length; i++) {
+        if (array[i] > max) {
+            max = array[i];
+        }
+    }
+    return max;
+}
+
+console.log(maxNumberOfArr(arrayOne)); // 44
+console.log(maxNumberOfArr(arrayTwo)); // -1
+console.log(maxNumberOfArr(arrayThree)); // 7
+console.log(maxNumberOfArr(arrayFour)); // 5
+console.log(maxNumberOfArr(arrayFive)); // 5
+
+function minNumberOfArr(array) {
+    let min = array[0];
+    for (let i = 1; i < array.length; i++) {
+        if (array[i] < min) {
+            min = array[i];
+        }
+    }
+    return min;
+}
+
+console.log(minNumberOfArr(arrayOne)); // -12
+console.log(minNumberOfArr(arrayTwo)); // -8
+console.log(minNumberOfArr(arrayThree)); // 1
+console.log(minNumberOfArr(arrayFour)); // -3
+console.log(minNumberOfArr(arrayFive)); // -3
+//
+function sumNumbersOfArr(array) {
+    let sum = 0;
+    for (let i = 0; i < array.length; i++) {
+        if (Number.isNaN(array[i])) {
+            sum += array[i];
+        }
+    }
+    return sum;
+}
+
+console.log(sumNumbersOfArr(arrayOne)); // 32
+console.log(sumNumbersOfArr(arrayTwo)); // -11
+console.log(sumNumbersOfArr(arrayThree)); // 11
+console.log(sumNumbersOfArr(arrayFour)); // 6
+console.log(sumNumbersOfArr(arrayFive)); // 6
+
+function onlyNegativeNumbers(array) {
+    return array.filter((e) => e < 0);
+}
+
+console.log(onlyNegativeNumbers(arrayOne)); // [-5, -12, -3, -3, -2, -3, -1]
+console.log(onlyNegativeNumbers(arrayTwo)); // [-1, -8, -2]
+console.log(onlyNegativeNumbers(arrayThree)); // []
+console.log(onlyNegativeNumbers(arrayFour)); //  [-3]
+console.log(onlyNegativeNumbers(arrayFive)); // [-3]
+
+function onlyPositiveNumbers(array) {
+    return array.filter((e) => e > 0);
+}
+
+console.log(onlyPositiveNumbers(arrayOne));// [3, 1, 44, 3, 1, 2, 2, 1, 4]
+console.log(onlyPositiveNumbers(arrayTwo)); // [];
+console.log(onlyPositiveNumbers(arrayThree)); // [1, 7, 3];
+console.log(onlyPositiveNumbers(arrayFour)); // [1, 3, 5];
+console.log(onlyPositiveNumbers(arrayFive)); // [1, 3, 5];
+
+// custom reduce function //
+function customReduce(array, reducer, initialValue) {
+    let accumulator = initialValue;
+    for (let i = 0; i < array.length; i++) {
+        accumulator = reducer(accumulator, array[i]);
+    }
+    return accumulator;
+}
+const numbers = [34, 42, 12, 15];
+
+const sum = customReduce(numbers, (accumulator, number) => accumulator + number, 0);
+
+console.log(sum); // 103
+
+// custom map function //
+
+function customMap(array, mapper) {
+    const result = [];
+    for (let i = 1; i < array.length; i++) {
```

Comment:
The loop should start from i = 0 instead of i = 1 because array indices start from 0
<img width="1393" alt="Screenshot 2023-11-21 at 13 25 05" src="https://github.com/MastersAcademy/fe-course-basic-2023/assets/118203044/1c07a0b2-ba16-4ae0-b34c-9ce75e28f2ef">